### ISSUE TITLE
cleanup sql panic: nid must be not nil

### ISSUE THREAD

**Issue by hyperknot** [ory/kratos#3030]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/kratos/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/kratos/blob/master/CONTRIBUTING.md).
- [X] This issue affects my [Ory Network](https://www.ory.sh/) project.
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [X] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

Running `cleanup sql` panics.

### Reproducing the bug

quickstart.yml is the https://github.com/ory/kratos/blob/master/contrib/quickstart/kratos/email-password/kratos.yml file.

```
export DSN="sqlite://db.sqlite?_fk=true&mode=rwc"

rm -f db.sqlite
kratos migrate sql -c config/quickstart.yml -e --yes
kratos cleanup sql -c config/quickstart.yml
```

The same happens on brew installed as well as locally compiled version.

### Relevant log output

```shell
Cleaning up expired sessions                 
audience=application service_name=Ory Kratos service_version=v0.11.1
panic: nid must be not nil [recovered]
  panic: nid must be not nil

goroutine 1 [running]:
github.com/ory/x/otelx.End({0x103e17680?, 0x104b6fea0}, 0x14001187888)
  /go/pkg/mod/github.com/ory/x@v0.0.531/otelx/withspan.go:52 +0x118
panic({0x103b52c40, 0x103dfc6b8})
  /usr/local/go/src/runtime/panic.go:884 +0x204
github.com/ory/x/contextx.(*Default).Network(0x1400066a340?, {0x14000431400?, 0x98?}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...})
  /go/pkg/mod/github.com/ory/x@v0.0.531/contextx/default.go:20 +0x16c
github.com/ory/kratos/persistence/sql.(*Persister).NetworkID(0x14000cc1da0, {0x103e0fa18, 0x14000d32210})
  /project/persistence/sql/persister.go:72 +0x7c
github.com/ory/kratos/persistence/sql.(*Persister).DeleteExpiredSessions(0x14000cc1da0, {0x103e0f9a8, 0x140000480b0}, {0x0?, 0x14001187b10?, 0x104b3ebc0?}, 0x1400061c5b4?)
  /project/persistence/sql/persister_session.go:406 +0x17c
github.com/ory/kratos/persistence/sql.(*Persister).CleanupDatabase(0x14000cc1da0, {0x103e0f9a8, 0x140000480b0}, 0x103e06c50?, 0x0, 0x140000107d0?)
  /project/persistence/sql/persister.go:147 +0x144
github.com/ory/kratos/cmd/cliclient.(*CleanupHandler).CleanupSQL(0x0?, 0x14000ad2c00, {0x14000a50980, 0x0, 0x0?})
  /project/cmd/cliclient/cleanup.go:61 +0x430
github.com/ory/kratos/cmd/cleanup.NewCleanupSQLCmd.func1(0x14000ad2c00?, {0x14000a50980?, 0x2?, 0x2?})
  /project/cmd/cleanup/sql.go:35 +0x38
github.com/spf13/cobra.(*Command).execute(0x14000ad2c00, {0x14000a50960, 0x2, 0x2})
  /go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x5c8
github.com/spf13/cobra.(*Command).ExecuteC(0x14000577b00)
  /go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x35c
github.com/spf13/cobra.(*Command).Execute(...)
  /go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/ory/kratos/cmd.Execute()
  /project/cmd/root.go:61 +0x24
main.main()
  /project/main.go:21 +0x64
```

### Version

master

### On which operating system are you observing this issue?

macOS

### In which environment are you deploying?

Binary

**Comment by frederikhors** [comment]
Same here.

**Comment by kelkarajay** [comment]
The raw query expects network ID for the delete operation but the CMD omits network ID during setup because of the option `skipNetworkInit`.

```go
	if o.skipNetworkInit {
		m.persister = p
		return nil
	}

	net, err := p.DetermineNetwork(ctx)
	if err != nil {
		m.Logger().WithError(err).Warnf("Unable to determine network, retrying.")
		return err
	}

	m.persister = p.WithNetworkID(net.ID)
	return nil
```

```go
func (p *Persister) DeleteExpiredSessions(ctx context.Context, expiresAt time.Time, limit int) (err error) {
	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.DeleteExpiredSessions")
	defer otelx.End(span, &err)
	err = p.GetConnection(ctx).RawQuery(fmt.Sprintf(
		"DELETE FROM %s WHERE id in (SELECT id FROM (SELECT id FROM %s c WHERE expires_at <= ? and nid = ? ORDER BY expires_at ASC LIMIT %d ) AS s )",
		"sessions",
		"sessions",
		limit,
	),
		expiresAt,
		p.NetworkID(ctx),
	).Exec()
	if err != nil {
		return sqlcon.HandleError(err)
	}
	return nil
}
```

@aeneasr

**Comment by aeneasr** [comment]
nice :)

**Comment by frederikhors** [comment]
@aeneasr will it be available for 0.11.2?

**Comment by aeneasr** [comment]
yes
